test/functional/test_framework/wallet.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""A limited-functionality wallet, which may replace a real wallet in tests"""
+
+from test_framework.address import ADDRESS_BCRT1_P2WSH_OP_TRUE
+from test_framework.messages import (
+    COIN,
+    COutPoint,
+    CTransaction,
+    CTxIn,
+    CTxInWitness,
+    CTxOut,
+)
+from test_framework.script import (
+    CScript,
+    OP_TRUE,
+)
+from test_framework.util import (
+    assert_equal,
+    hex_str_to_bytes,
+    satoshi_round,
+)
+from decimal import Decimal
+
+
+class MiniWallet:
+    def __init__(self, test_node):
+        self._test_node = test_node
+        self._utxos = []
+        self._address = ADDRESS_BCRT1_P2WSH_OP_TRUE
+        self._scriptPubKey = hex_str_to_bytes(self._test_node.validateaddress(self._address)['scriptPubKey'])
+
+    def generate(self, num_blocks):
+        blocks = self._test_node.generatetoaddress(num_blocks, self.getnewaddress())
+        for b in blocks:
+            cb_tx = self._test_node.getblock(b, 2)['tx'][0]
+            self._utxos.append({'txid': cb_tx['txid'], 'vout': 0, 'value': cb_tx['vout'][0]['value']})
```

Comment:
Seems like the "wallet" doesn't know which funds are mature?

Reply:
The wallet will sign any transaction you ask it to sign. Also, coin selection is deterministic, so if immature funds are spent, it will be noticed when writing the test.

Reply:
I think the real answer is you're relying on FIFO which picks oldest first.

Reply:
The wallet doesn't currently query the chain in any way, so it has no notion of number of confirmations.

If this is ever needed, it can be added later